java/dagger/internal/codegen/Binding.java

```diff
@@ -130,6 +139,11 @@ final FrameworkType frameworkType() {
                   .map(DependencyAssociation::frameworkDependency)
                   .collect(toImmutableList()));
 
+  // TODO this would be better optimized to share the parsed metadata for all bindings originating
+  //  from a given contributing module
+  private final Supplier<Optional<KmClass>> moduleKmClass =
+      memoize(() -> kmClassOf(contributingModule()));
```

Comment:
This should all probably be in `ContributionBinding`. `MembersInjectionBinding` never has a module.

Reply:
Will move 👍